test/performance/search/index_construction_benchmark.cpp

```diff
@@ -0,0 +1,132 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+// written for the purpose of the Bch Softwarepraktikum, early 2019 FU Berlin
+// @author: Clemens Cords <clemenscords@fu-berlin.de>
+
+#include <cstring>
+#include <random>
+
+#include <benchmark/benchmark.h>
+
+#include <seqan3/io/stream/debug_stream.hpp>
+#include <seqan3/search/fm_index/all.hpp>
+#include <seqan3/alphabet/all.hpp>
+
+using namespace seqan3;
+
+const std::size_t LENGTH = 50;
+const unsigned int SEED = 1234;
+
+// ============================================================================
+// generate sequence deterministically based on fixed const seed
+// ============================================================================
+template<Alphabet alphabet_t>
+std::vector<alphabet_t> generate_test_sequence(const std::size_t length)
+{
+    const size_t alphabet_size = alphabet_t::value_size;
+
+    std::mt19937 rng_engine;
+    rng_engine.seed(SEED);
+    std::uniform_int_distribution distribution{0, static_cast<int>(alphabet_size-1)};
+
+    std::vector<alphabet_t> text{};
+
+    for (std::size_t i = 0; i < length; ++i)
+    {
+        alphabet_t s{};
+        s.assign_rank(distribution(rng_engine));
+        text.push_back(s);
+    }
+
+    return text;
+}
+
+// ============================================================================
+// "overload" for chars
+// ============================================================================
+std::vector<char> generate_test_sequence_char(const std::size_t length)
```

Comment:
hm can you try if our function in the code base works for `char` ? 
(Exchange the member function `assign_rank` with the free function before you try it)
line 41 should be:
```cpp
        sequence.push_back(assign_rank(alphabet_t{}, dis_alpha(gen)));
```
I think assign rank is overloaded for `char`

Reply:
I edited ``seqan3/test/include/seqan3/test/performance/sequence_generator.hpp`` and added it to this pull request since assign_rank is indeed overloaded for chars